esp/public/media/scripts/program/modules/scheduling_checks.jsx

```diff
@@ -114,3 +160,98 @@ var RefreshButton = React.createClass({
     </button>;
   },
 });
+
+/**
+ * Calls its onClick prop to reset table greying/sorting
+ */
+var ResetButton = React.createClass({
+  propTypes: {
+    onClick: React.PropTypes.func.isRequired,
+  },
+
+  render: function () {
+    return <button onClick={this.props.onClick} className="reset-button">
+      Reset
+    </button>;
+  },
+});
+
+// Modified from react-json-table example code.
+var SelectTable = React.createClass({
+  
+  propTypes: {
+    rows: React.PropTypes.array.isRequired,
+    saveState: React.PropTypes.shape({
+      greyed: React.PropTypes.object.isRequired,
+      sort: React.PropTypes.any.isRequired,
+      reverse: React.PropTypes.bool.isRequired
+    }).isRequired,
+    header: React.PropTypes.bool.isRequired,
+    columns: React.PropTypes.array,
+    updateTableState: React.PropTypes.func.isRequired
+  },
+    
+  getInitialState: function(){
+    // We will store the sorted column and whether each row is greyed out
+    return {sort: this.props.saveState.sort, greyed: this.props.saveState.greyed, reverse: this.props.saveState.reverse};
+  },  
+  render: function(){
+    this.props.updateTableState(this.state);
+    var me = this;
+    // clone the rows
+    items = this.props.rows.slice();
+    
+    items = _.sortBy(items, me.state.sort);
+    
+    if (this.state.reverse) items.reverse();
+    
+    return <JsonTable 
+      rows={items} 
+      columns={this.props.columns}
+      settings={ this.getSettings() } 
+      onClickHeader={ this.onClickHeader }
+      onClickRow={ this.onClickRow }
+    />;
+  },
+  
+  getSettings: function(){
+      var me = this;
+      // We will add some classes to the selected rows and cells
+      return {
+        headerClass: function( current, key ){
+            if( me.state.sort == key ) {
+              if ( me.state.reverse) {
+                return current + ' headerSelected sortReversed';
+              } else {
+                return current + ' headerSelected';
+              }
+            } else {
+              return current;
+            }
+        },
+        rowClass: function( current, item ){
+          if( me.state.greyed[item] ) {
+            return current + ' rowGreyed';
+          } else {
+            return current;
+          }
+        },
+        header: this.props.header
+      };
+  },
+    
+  onClickHeader: function( e, column ){
```

Comment:
I did that initially; it didn't work. The parent gets passed an outdated version of the state (in particular, the most recent state change isn't processed). There's a thing on https://facebook.github.io/react/docs/component-api.html which seems to be relevant, about how it doesn't immediately mutate the state.